haystack/document_stores/pinecone.py

```diff
@@ -280,33 +404,73 @@ def get_document_count(
                     }
                 }
                 ```
-        :param index: Optional index to use for the query. If not provided, the default index is used.
+        :param index: Optional index name to use for the query. If not provided, the default index name is used.
         :param only_documents_without_embedding: If set to `True`, only documents without embeddings are counted.
         :param headers: PineconeDocumentStore does not support headers.
+        :param namespace: Optional namespace to count documents from. If not specified, None is default
+        :param type_metadata: Optional value for `doc_type` metadata to reference documents that need to be counted.
+            Parameter options:
+                - `"vector"`: Documents with embedding.
+                - `"no-vector"`: Documents without embedding (dummy embedding only).
+                - `"label"`: Labels.
         """
         if headers:
             raise NotImplementedError("PineconeDocumentStore does not support headers.")
 
-        index = self._index_name(index)
-        if index not in self.pinecone_indexes:
-            raise PineconeDocumentStoreError(
-                f"Index named '{index}' does not exist. Try reinitializing PineconeDocumentStore() and running "
-                f"'update_embeddings()' to create and populate an index."
-            )
+        # add `doc_type` value if specified, otherwise default is related to documents without embeddings
+        types_metadata = {type_metadata or self.document_without_embedding_metadata}
+        if not type_metadata and not only_documents_without_embedding:
+            # add `doc_type` related to documents with embeddings
+            types_metadata.add(self.document_with_embedding_metadata)
 
-        pinecone_syntax_filter = LogicalFilterClause.parse(filters).convert_to_pinecone() if filters else None
```

Comment:
this is removed because we apply the filters in `_get_vector_count`?

Reply:
That's right. Common logic from `get_embedding_count` and `get_document_count` is extracted in helper method `_get_vector_count`. First step in `_get_vector_count` is to extend given filters with new ones related to `doc_type` metadata field using `types_metadata` values (if any) and after that to do parsing into Pinecone syntax.